### PR TITLE
add default chrome capabilities only for chrome. Fix https://github.c…

### PR DESCRIPTION
**Pull request by Taxanas**
…om/fhoeben/hsac-fitnesse-fixtures/issues/284

### CODE REVIEW

**Comment by fhoeben** [comment]
Does it make sense to invert the check: instead of only setting for chrome NOT set for Android (and maybe iOS, have you tested that)?
Since it is part of the W3C spec I would want to set it for all browsers, and not for those we find do not support it.

**Comment by fhoeben** [comment]
Does it make sense to invert the check: instead of only setting for chrome NOT set for Android (and maybe iOS, have you tested that)?
Since it is part of the W3C spec I would want to set it for all browsers, and not for those we find do not support it.

**Comment by Taxanas** [comment]
I agree, and will try it tomorrow 😉 
Actually, current Appium logs unknown cappabilities, ignores them, and works just fine.
My suggested PR is more of a workaround to connectToAndroidDriverAtWithCapabilities method adding those into desiredCapabilities and copying into capabilities in mentioned jsons. 
P. S. I got myself lost in debuging differences between connectToAndroidDriverAtWithCapabilities and connectToDriverAtWithCapabilities. 🙂

P. P. S. As i have an opportunity to communicate to you, might i ask whether there are plans for connectToWindowsDriverAtWithCapabilities? 🙂

**Comment by fhoeben** [comment]
@tcnh didn't you once mention you created an extension to connect to Windows devices? Would that be something like the `connectToWindowsDriverAtWithCapabilities` @Taxanas is looking for? If so, any chance you can turn that into a PR (or did you already publish it yourself somewhere)?

**Comment by Taxanas** [comment]
Hello, @fhoeben 
Wanted to ask your opinion on this:
https://github.com/fhoeben/hsac-fitnesse-fixtures/pull/285/commits/d4357146144a03370758023646b1b62a0470f0e3 
:)

File: src/main/java/nl/hsac/fitnesse/fixture/util/selenium/driverfactory/DriverFactory.java [hunk 1]
```diff
@@ -5,14 +5,18 @@
...
+
 /**
  * Interface to encapsulate creation of Selenium driver.
  */
 public interface DriverFactory {
     WebDriver createDriver();
 
     static void addDefaultCapabilities(MutableCapabilities capabilities) {
-        if (!capabilities.getCapabilityNames().contains(CapabilityType.UNEXPECTED_ALERT_BEHAVIOUR)) {
+        Set<String> capabilitieNames = capabilities.getCapabilityNames();
```
Comment (fhoeben) [inline src/main/java/nl/hsac/fitnesse/fixture/util/selenium/driverfactory/DriverFactory.java]:
```suggestion
        Set<String> capabilityNames = capabilities.getCapabilityNames();
```

**Comment by tcnh** [comment]
I have to say I'm not completely up to date on appium right now. We decided to start an initiative to decouple windows fixture implementation from the mobile device fixtures, because the interaction is different.  I don't exactly know the curent status but I know some of our consultants are still effectively using https://repo.maven.apache.org/maven2/nl/praegus/toolchain-appium-fixtures/0.0.12/ (https://github.com/praegus/toolchain-appium-fixtures) for android, iOS and Windows 10 and have done some fixes on it recently, so maybe it contains what you need?

**Comment by Taxanas** [comment]
Hello, @fhoeben ,

Today i tried connectToAndroidDriverAtWithCapabilities from praegus/toolchain-appium-fixtures. Same issue appears when trying to connect to grid.
@tcnh , thank you for sharing info :)

Any updates about pull request?

**Comment by tcnh** [comment]
That's strange, I checked with a few colleagues that use toolchain-appium-fixtures.. none of them experience issues connecting with appium. Are you trying to get an appium session on a selenium grid? Because I think that will not work...? If that should work, I can look into what needs to change

**Comment by Taxanas** [comment]
@tcnh ,

> Are you trying to get an appium session on a selenium grid?
Exactly ;) i have various browsers, os **and mobile devices** connected to grid via appium.

When i connect directly to node, everything is _ok_, but when i request for node via grid, it tries to parse json and fails :) 

> Because I think that will not work...?
I have already tested it ;) works smooth, **but**...  **unexpectedAlertBehaviour** and **unhandledPromptBehavior** should stay under desiredCapabilites, not in capabilities in json, that is passed to GRID when requiring for node, same way, that it is requested when using simple **connectToDriverAtWithCapabilities** from hsac fixtures **OR** preferably would not be added to json at all when not requiring browser to work with :)

More about it:
https://github.com/fhoeben/hsac-fitnesse-fixtures/issues/284

**Comment by tcnh** [comment]
Ah, I get what the issue is now. 

Funny thing is that when using a local driver, addDefaultCapabilities is _only_ called for chrome. When using a remote driver, it's always called. That's not completely consistent (and some indication that these caps are not widely accepted?)

I think setting only for chrome is the sensible solution, but will that work when testing chrome _on_ an actual android device?

**Comment by Taxanas** [comment]
@tcnh , that is correct ;)
If you ask for chrome browser on android, it would still be requested by _browserName_, right? ;)
Thats why i don't want to emit it by platformName ;)

Latest commit https://github.com/fhoeben/hsac-fitnesse-fixtures/pull/285/commits/bcc6fc55488e390654720a31a0e12f4ab004fdaf

Makes them being added whenever browser is requested from GRID :)

**Pull request merged**